sqlmesh/core/model/cache.py

```diff
@@ -100,15 +102,11 @@ def with_optimized_query(self, model: Model, name: t.Optional[str] = None) -> bo
         cache_entry = self._file_cache.get(name)
         if cache_entry:
             try:
-                if cache_entry.optimized_rendered_query:
-                    model._query_renderer.update_cache(
-                        cache_entry.optimized_rendered_query, optimized=True
-                    )
-                else:
-                    # If the optimized rendered query is None, then there are likely adapter calls in the query
-                    # that prevent us from rendering it at load time. This means that we can safely set the
-                    # unoptimized cache to None as well to prevent attempts to render it downstream.
-                    model._query_renderer.update_cache(None, optimized=False)
+                # If the optimized rendered query is None, then there are likely adapter calls in the query
+                # that prevent us from rendering it at load time. This means that we can safely set the
+                # unoptimized cache to None as well to prevent attempts to render it downstream.
+                optimized = cache_entry.optimized_rendered_query is not None
+                model._query_renderer.update_cache(cache_entry, optimized=optimized)
```

Comment:
Let's not leak `OptimizedQueryCacheEntry` outside `cache.py`. I suggest passing `optimized_rendered_query` and `renderer_violations` as separate arguments.